### DISCUSSION TITLE
[Question] How to pass a value or datas in the callback function of nng_aio_alloc?

### DISCUSSION THREAD

**Discussion by alawn-wang** [nanomsg/nng#1799] [Q&A] [answered]
I have create a server to listen the TCP stream connect, such like `nng_stream_listener_accept(listener, aio_accept)`, when a stream connected, the callback function of `nng_aio* aio_accept` will be called, i want to pass some data in this function. Is there any way to implement it?

**Comment by gdamore** [comment, marked as answer]
So typically you set this up with `nng_aio_alloc()` -- the intention is that the `void *` parameter is one you (the consumer) supply, which can then be passed to the callback (in fact it's the first argument to the callback.)

It is possible to cheat and use the input or output parameters for a little bit of data (e.g. `nng_aio_set_input()` with a value of 4 or so, but generally speaking this is risky because the input and output callers are "owned" by the stack in the middle.  Usually they only use the first 1 or 2.

(At one point I had an API for attaching other data to AIOs, but in very large scale systems this causes the memory use to bigger than I'd like, and its not strictly necessary with the callback parameter, so I got rid of it.)

**Reply by alawn-wang** [reply]
ok, Thanks for your reply. I seems that I can only pass parameter to callback function.  But can't pass data from callback to AIO.
